libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Ast.hs

```diff
@@ -230,7 +230,13 @@ data RootTarget era root t where
   (:$) :: RootTarget era r (a -> b) -> RootTarget era r a -> RootTarget era r b
   Constr :: String -> (a -> b) -> RootTarget era Void (a -> b)
   Invert :: String -> TypeRep root -> (a -> b) -> RootTarget era root (a -> b)
-  Lensed :: Term era t -> Lens' root t -> RootTarget era root t
+  Lensed :: Term era t -> SimpleGetter root t -> RootTarget era root t
+  Partial ::
+    Term era t ->
+    (root -> Maybe t) ->
+    -- | Partial designed to be used in Pred 'Oneof' where the RootTarget getter may fail.
+    --   So use 'Partial' instead of 'Lensed' in 'Oneof'
+    RootTarget era root t
```

Comment:
So the whole `Partial` is needed because some of the  data constructors of GovAction have the `StrictMaybe (PrevGovActionId)` parameter which needs to have specific `GovActionPurpose`   ?

Reply:
So a RootTarget is supposed to be invertible. Given a solution root for a RootTarget and an (Env era) get back a new (Env era) that contains bindings for the variables in the RootTarget extracted from the solution.
```
getTarget :: forall era root t. root -> RootTarget era root t -> Env era -> Env era
```
A RootTarget with Partial might not return an Env

```
getTargetPartial :: forall era root t. root -> RootTarget era root t -> Env era -> Maybe(Env era)
```

This makes it usefull for generating Sum types.